src/lib/group.test.ts

```diff
@@ -1,30 +1,46 @@
-import {
-  shutdown,
-  isReady,
-  Field,
-  Bool,
-  Circuit,
-  Group,
-  Scalar,
-  Provable,
-} from 'snarkyjs';
+import { Bool, Group, Scalar, Provable } from 'snarkyjs';
 
 describe('group', () => {
   let g = Group({
     x: -1,
     y: 2,
   });
-  beforeAll(async () => {
-    await isReady;
-  });
-
-  afterAll(async () => {
-    setTimeout(async () => {
-      await shutdown();
-    }, 0);
-  });
 
   describe('Inside circuit', () => {
```

Comment:
can we use the quickcheck generator Gregor made earlier to check some of the algebraic properties for edge cases?

Reply:
I was thinking about that as well, however, I think there is no easy way to sample valid Group elements other than brute forcing (or is there?) that's why the existing tests mostly work with the generator and `(-1, 2)`

Reply:
I used to sample public keys (= group elements) by picking a random private key (= scalar) and scaling the group generator with it.
That's not super efficient but good enough.

However, now we have hashToCurve which should be faster, especially if you skip the hash and only do random field element + mapToCurve

Reply:
Oh right, I didn't consider that at all :face_with_spiral_eyes: